build/main.go

```diff
@@ -92,6 +99,22 @@ type NativeAmount struct {
 	Amount string
 }
 
+type Path struct {
+	Assets []Asset
+}
+
+// PathDestination is a mutator that configures a path_payment's destination asset and amount
+type PathDestination struct {
```

Comment:
> I'm only referring to the `PathDestination` struct in the comment above... It doesn't deal source portion of the payment.

There's `PathSend` struct below. It all makes sense, I was saying that with the current design it wasn't possible to set both SendMax and DestAmount the API I created was one of possible workarounds. I like your new design but I'd change one thing. Check my previous comment added to your snippet.